pkg/collector/corechecks/cluster/kubernetesapiserver/bundled_events.go

```diff
@@ -59,13 +60,21 @@ func (c *bundledTransformer) Transform(events []*v1.Event) ([]event.Event, []err
 
 		id := buildBundleID(event)
 
-		bundle, found := bundlesByObject[id]
+		bundles, found := bundlesByObject[id]
 		if !found {
-			bundle = newKubernetesEventBundler(c.clusterName, event)
-			bundlesByObject[id] = bundle
+			bundles = []*kubernetesEventBundle{newKubernetesEventBundler(c.clusterName, event)}
+			bundlesByObject[id] = bundles
 		}
 
-		err := bundle.addEvent(event)
+		lastBundle := bundles[len(bundles)-1]
+		_, fits := lastBundle.fitsEvent(event)
+		if !fits {
+			bundles = append(bundles, newKubernetesEventBundler(c.clusterName, event))
+			lastBundle = bundles[len(bundles)-1]
```

Comment:
I think here the `lastBundle` will always be the one you just created L72 with `newKubernetesEventBundler` no ? 
could do soemthing like:

```go
lastBunfle = newKubernetesEventBundler(c.clusterName, event)
bundles = append(bundles, lastBundles)
``` 
?